oxd-server/src/main/java/org/gluu/oxd/server/persistence/service/GluuPersistenceService.java

```diff
@@ -0,0 +1,306 @@
+                return null;
+            }
+            LOG.error("Failed to fetch ExpiredObject by id: {} ", key, e);
+        }
+        return null;
+    }
+
+    private ExpiredObject getExpiredObject(String key, String... returnAttributes) {
+        return (ExpiredObject) this.persistenceEntryManager.find(getDnForExpiredObj(key), ExpiredObject.class, returnAttributes);
+    }
+
+    public boolean isExpiredObjectPresent(String key) {
+        return getExpiredObject(key) != null;
+    }
+
+    public boolean removeAllRps() {
+        try {
+            this.persistenceEntryManager.remove(String.format("%s,%s", new Object[]{getRpOu(), getOxdDn()}), RpObject.class, null, 100);
+            LOG.debug("Removed all Rps successfully. ");
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to remove all Rps", e);
+        }
+        return false;
+    }
+
+    public Set<Rp> getRps() {
+        try {
+            List<RpObject> rpObjects = this.persistenceEntryManager.findEntries(String.format("%s,%s", new Object[]{getRpOu(), getOxdDn()}), RpObject.class, null);
+
+            Set<Rp> result = new HashSet();
+            for (RpObject ele : rpObjects) {
+                Rp rp = MigrationService.parseRp(ele.getData());
+                if (rp != null) {
+                    result.add(rp);
+                } else {
+                    LOG.error("Failed to parse rp, id: {}, dn: {} ", ele.getId(), ele.getDn());
+                }
+            }
+            return result;
+        } catch (Exception e) {
+            if (((e instanceof EntryPersistenceException)) && (e.getMessage().contains("Failed to find entries"))) {
+                LOG.warn("Failed to fetch RpObjects. {} ", e.getMessage());
+                return null;
+            }
+            LOG.error("Failed to fetch rps. Error: {} ", e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public void destroy() {
+        this.persistenceEntryManager.destroy();
+    }
+
+    public boolean remove(String oxdId) {
+        try {
+            this.persistenceEntryManager.remove(getDnForRp(oxdId));
+
+            LOG.debug("Removed rp successfully. oxdId: {} ", oxdId);
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to remove rp with oxdId: {} ", oxdId, e);
+        }
+        return false;
+    }
+
+    public boolean deleteExpiredObjectsByKey(String key) {
+        try {
+            this.persistenceEntryManager.remove(getDnForExpiredObj(key));
+            LOG.debug("Removed expired_objects successfully: {} ", key);
+            return true;
+        } catch (Exception e) {
+            LOG.error("Failed to remove expired_objects: {} ", key, e);
+        }
+        return false;
+    }
+
+    public boolean deleteAllExpiredObjects() {
+        try {
+            this.persistenceEntryManager.remove(String.format("%s,%s", new Object[]{getExpiredObjOu(), getOxdDn()}), ExpiredObject.class, null, 100);
```

Comment:
We need configuration property `persistence-manager-remove-count: 1000`

Reply:
Done